cloudbuild.yaml

```diff
@@ -23,25 +23,28 @@ steps:
   # build gke-exec-auth-plugin binary
   - name: 'gcr.io/cloud-builders/bazel'
     args:
+      - --output_base=/workspace/plugin-binaries
```

Comment:
FWIW I would recommend that we put the entire bazel cache under `/workspace` in _all_ steps so we actually get some benefit from bazel.

Right now every step is having to compile everything from scratch.

I think we should use `--output_user_root=/workspace/bazel-root` (includes unpacked bazel install) and `--output_base=/workspace/bazel-base` in all steps.

Reply:
we can see builds are getting slower, so might want to prioritize ensuring the steps share the cache